src/helpers.ts

```diff
@@ -4,41 +4,39 @@
  *--------------------------------------------------------------------------------------------*/
 import * as fs from "fs";
 import * as JSONC from "jsonc-parser";
+import Mustache from "mustache";
 import * as log from "./Log";
+import ISettingsConfigJson from "./types/ISettingsConfigJson";
 
-/**
- * Loads a settings file and validates it with JSON schema, then returns it as a typed object.
- * @param serviceName The name of the service loading the settings. Used in log messages.
- * @param settingsFileName The path to the file to load.
- * @type T The type the settings should return as.
- */
-export function readSettings<T>(serviceName: string, settingsFileName: string): T {
-  let rawConfig: string;
+function readFile(filePath: string, fileType: string, serviceName: string): string {
   try {
-    rawConfig = fs.readFileSync(settingsFileName, "utf-8");
+    return fs.readFileSync(filePath, "utf-8");
   } catch (e) {
-    log.warn(serviceName, `Unable to read the configuration file: ${e.message}.`);
+    log.warn(serviceName, `Unable to read the ${fileType} file: ${e.message}.`);
     return null;
   }
+}
 
-  // This shouldn't happen. Keeping the check here in case it does in the real world
-  // and someone reports things not working.
-  if (!rawConfig) {
-    throw new Error(`[${serviceName}] Unable to load configuration file ${settingsFileName}.`);
-  }
-
-  const parseErrors: JSONC.ParseError[] = [];
+function parseFile(filePath: string, fileType: string, serviceName: string) {
+  const file = readFile(filePath, fileType, serviceName);
+  return file ? JSONC.parse(file) : null;
+}
 
-  const settings = JSONC.parse(rawConfig, parseErrors) as T;
+function replaceSecrets(settings: ISettingsConfigJson, secrets: string) {
```

Comment:
This needs doc comments on it.

Reply:
So, just some thoughts I'd like to add. This is totally up to you, since this is your repo.

Arguably the best development book I've read [Clean Code](https://www.amazon.com/dp/B001GSTOAM), advocates for no comments. They don't really talk about docs IIRC, but I advocate for only high level documentation. If we need to know how the functionality of the system works, rely on the tests to tell that story. In other words, the tests are the docs, write them in a readable and very easily understandable way, write them to cover everything, and if TDD is being done properly (test first), then your system under tests (production code) will change as your tests drive them to change, and also your docs (the tests) will stay up to date as a result. That being said, consumers of the package still need to know what the package does as a whole and that doesn't typically change, so it's ok to document that part outside of the code.

What are your thoughts on this? The easy solution is for me to just add a doc comment to the function and move on, but I'm here to do more than that if you're open to it 🤷 .

Reply:
I haven't read Clean Code but have read Refactoring: Improving the Design of Existing Code. It advocates the same: name your methods by what they do.

The reason I want doc comments on everything is for Intellisense. Editors like Visual Studio Code will pick them up and show them while you are writing code that uses the method.

Reply:
Updated